### PR TITLE
[python] added f-strings to docs/conf.py

### PR DESCRIPTION
**Pull request by NovusEdge**

### CODE REVIEW

**Comment by NovusEdge** [comment]
#4136

**Comment by NovusEdge** [comment]
@StrikerRUS 
please do let me know about any changes to be made in the split PRs

**Comment by jameslamb** [comment]
> @StrikerRUS
> please do let me know about any changes to be made in the split PRs

Please see my suggestions at https://github.com/microsoft/LightGBM/pull/4142#pullrequestreview-624538246. These apply to all of the PRs you've opened, and I think will help you debug locally.

File: docs/conf.py [hunk 1]
```diff
@@ -206,10 +206,10 @@ def generate_doxygen_xml(app):
     app : object
         The application object representing the Sphinx process.
     """
+    INPUT = os.path.join(CURR_PATH, os.path.pardir, 'include', 'LightGBM', 'c_api.h')
```
Comment (StrikerRUS) [inline docs/conf.py:209]: ALL_CAPS names are used for module-level variables. Inside a function we use snake_case naming schema for variables.
```suggestion
    input = os.path.join(CURR_PATH, os.path.pardir, 'include', 'LightGBM', 'c_api.h')
```

File: docs/conf.py [hunk 2]
```diff
@@ -206,10 +206,10 @@ def generate_doxygen_xml(app):
...
     doxygen_args = [
-        "INPUT={}".format(os.path.join(CURR_PATH, os.path.pardir,
-                                       'include', 'LightGBM', 'c_api.h')),
-        "OUTPUT_DIRECTORY={}".format(os.path.join(CURR_PATH, 'doxyoutput')),
+        f"INPUT={INPUT}",
```
Comment (StrikerRUS) [inline docs/conf.py:211]:
```suggestion
        f"INPUT={input}",
```

File: docs/conf.py [hunk 3]
```diff
@@ -206,10 +206,10 @@ def generate_doxygen_xml(app):
...
+        f"OUTPUT_DIRECTORY={ os.path.join(CURR_PATH, 'doxyoutput') }",
```
Comment (StrikerRUS) [inline docs/conf.py:212]:
```suggestion
        f"OUTPUT_DIRECTORY={os.path.join(CURR_PATH, 'doxyoutput')}",
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
@NovusEdge Thanks! Please take a look at me comments below:

**Review by StrikerRUS — APPROVED** [review]
LGTM!
I don't believe that macOS R-package jobs are failing due to the changes from this PR. I guess we can wait a little bit with a hope that issues will go away without our help.

**Comment by jameslamb** [comment]
Hopefully should be resolved soon! Looks like our old friend, the MiKTeX mirror.

```text
==> Downloading http://mirror.ctan.org/systems/mac/mactex/mactex-basictex-20200407.pkg
==> Downloading from https://ctan.math.illinois.edu/systems/mac/mactex/mactex-basictex-20200407.pkg
curl: (22) The requested URL returned error: 404 
Error: Download failed on Cask 'basictex' with message: Download failed: http://mirror.ctan.org/systems/mac/mactex/mactex-basictex-20200407.pkg
sudo: tlmgr: command not found
sudo: tlmgr: command not found
```

https://github.com/microsoft/LightGBM/pull/4147/checks?check_run_id=2254678263

**Comment by jameslamb** [comment]
I've updated this PR to latest `master` to get the CI fixes from #4161. Will merge if / when it builds successfully.

Thanks for the help with these @NovusEdge !

**Review by jameslamb — APPROVED** [review]

**Pull request merged**